Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/network/NetworkStateProvider.kt

```diff
@@ -74,8 +74,22 @@ class NetworkStateProvider @Inject constructor(
         get() = manager.activeNetwork.let { network ->
             State(
                 activeNetwork = network,
-                capabilities = network?.let { manager.getNetworkCapabilities(it) },
-                linkProperties = network?.let { manager.getLinkProperties(it) },
+                capabilities = network?.let {
+                    try {
+                        manager.getNetworkCapabilities(it)
+                    } catch (e: SecurityException) {
+                        Timber.tag(TAG).e(e, "Failed to determine network capabilities.")
+                        null
+                    }
+                },
+                linkProperties = network?.let {
+                    try {
+                        manager.getLinkProperties(it)
+                    } catch (e: Exception) {
+                        Timber.tag(TAG).e(e, "Failed to determine link properties.")
+                        null
+                    }
+                },
```

Comment:
is it possible to wrap both statements in one `try-catch` block?

Reply:
I currently don't see a way that would not make the code less readable.